novawallet/Modules/AssetList/Models/AssetListBuilderResult.swift

```diff
@@ -50,7 +50,16 @@ struct AssetListBuilderResult {
         }
 
         func hasSwaps() -> Bool {
-            allChains.values.contains { $0.hasSwaps }
+            balanceResults.contains {
```

Comment:
As we have much less chains then the balances, then I would iterate over the chains